### PR TITLE
Ensure that kernel IOPub is up before starting tests

### PR DESCRIPTION
**Pull request by vidartf**
We would sometimes see random timeouts on the first cell of a notebook waiting for an IOPub message. This would occur if the initial messages, including the idle message, was discarded due to the IOPub still not being up.

This PR proposes to fix this by polling the kernel with `kernel_info()` during start up until it receives a reply on IOPub. It also does some varying cleanup and needed refactoring in separate commits.

### CODE REVIEW

File: nbval/kernel.py [hunk 1]
```diff
@@ -100,11 +124,11 @@ def get_message(self, stream, timeout=None):
             elif stream == 'shell':
                 msg = self.kc.get_shell_msg(timeout=timeout)
             else:
-                raise ValueError('Invalid stream specified: "%s"' % stream)
+                raise ValueError('Invalid stream specified: "%s"', stream)
```
Comment (takluyver) [inline nbval/kernel.py]: This works for logging, but not for errors, I think.

**Comment by takluyver** [comment]
One minor thing to change, otherwise +1

File: nbval/kernel.py (hunk 1 above)
Comment (vidartf) [inline nbval/kernel.py]: Yes, I think that was me being a bit too fast :)

**Comment by vidartf** [comment]
Fixed the review point, and also deleted some unused files (they seem to be there as a historical record? This is kind of the job of the VCS).

**Comment by takluyver** [comment]
Thanks @vidartf

**Pull request merged**